hash/hash_comparator.go

```diff
@@ -199,3 +230,33 @@ func decodePbkdf2Hash(encodedHash string) (p *Pbkdf2, salt, hash []byte, err err
 
 	return p, salt, hash, nil
 }
+
+// decodeScryptHash decodes Scrypt encoded password hash.
+// format: $scrypt$N=<cost>,r=<block>,p=<parrrelization>$<salt>$<hash>
+func decodeScryptHash(encodedHash string) (p *Scrypt, salt, hash []byte, err error) {
+	parts := strings.Split(encodedHash, "$")
+	if len(parts) != 5 {
+		return nil, nil, nil, ErrInvalidHash
+	}
+
+	p = new(Scrypt)
+
+	_, err = fmt.Sscanf(parts[2], "N=%d,r=%d,p=%d", &p.Cost, &p.Block, &p.Parrellization)
```

Comment:
Are these parameters following a standard and if so where can I find it? :)

Reply:
I have not found a standard... how I am using the hashing now([in perl](https://metacpan.org/pod/Crypt::ScryptKDF)) the format is SCRYPT:N:R:P:<Salt>:<Hash>, since I could not find a standard, I just went with similar to how the other algorithms are done.

Reply:
I have found a [python library](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.scrypt.html) it is very similar to the format I suggested but different than the perl, I don't guess there is a standard

Reply:
Cool, I will change it to this way :)

Reply:
thank you!